### PR TITLE
refactor: replace TRUE/FALSE macros with C99 true/false

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/mbyte.c [hunk 1]
```diff
@@ -2354,10 +2354,10 @@ int convert_setup(vimconv_T *vcp, char_u *from, char_u *to)
   return convert_setup_ext(vcp, from, true, to, true);
 }
 
-/*
- * As convert_setup(), but only when from_unicode_is_utf8 is TRUE will all
- * "from" unicode charsets be considered utf-8.  Same for "to".
- */
+//
```
Comment (bfredl) [inline src/nvim/mbyte.c:2357]: Remove empty lines like this

File: src/nvim/normal.c [hunk 2]
```diff
@@ -8146,9 +8146,9 @@ static void nv_event(cmdarg_T *cap)
   }
 }
 
-/*
- * Return TRUE when 'mousemodel' is set to "popup" or "popup_setpos".
- */
+//
```
Comment (bfredl) [inline src/nvim/normal.c:8149]: `/// @Return` . Also delete empty lines.

File: src/nvim/mbyte.c (hunk 1 above)
Comment (dundargoc) [reply src/nvim/mbyte.c]: fixed

File: src/nvim/normal.c (hunk 2 above)
Comment (dundargoc) [reply src/nvim/normal.c]: fixed

File: src/nvim/ex_cmds_defs.h [hunk 3]
```diff
@@ -150,16 +150,16 @@ struct exarg {
   cmdidx_T cmdidx;              ///< the index for the command
   uint32_t argt;                ///< flags for the command
   int skip;                     ///< don't execute the command, only parse it
-  int forceit;                  ///< TRUE if ! present
+  int forceit;                  ///< true if ! present
```
Comment (vigoux) [inline src/nvim/ex_cmds_defs.h:153]: Should be a `bool` then.

File: src/nvim/ex_cmds_defs.h [hunk 4]
```diff
@@ -150,16 +150,16 @@ struct exarg {
...
   linenr_T line1;               ///< the first line number
   linenr_T line2;               ///< the second line number or count
   cmd_addr_T addr_type;         ///< type of the count/range
   int flags;                    ///< extra flags after count: EXFLAG_
   char_u      *do_ecmd_cmd;     ///< +command arg to be used in edited file
   linenr_T do_ecmd_lnum;        ///< the line number in an edited file
-  int append;                   ///< TRUE with ":w >>file" command
-  int usefilter;                ///< TRUE with ":w !command" and ":r!command"
+  int append;                   ///< true with ":w >>file" command
+  int usefilter;                ///< true with ":w !command" and ":r!command"
```
Comment (vigoux) [inline src/nvim/ex_cmds_defs.h:162]: Should ne bools here too.

File: src/nvim/ex_cmds_defs.h [hunk 5]
```diff
@@ -195,7 +195,7 @@ struct expand {
   sctx_T xp_script_ctx;         // SCTX for completion function
   int xp_backslash;             // one of the XP_BS_ values
 #ifndef BACKSLASH_IN_FILENAME
-  int xp_shell;                 // TRUE for a shell command, more
+  int xp_shell;                 // true for a shell command, more
```
Comment (vigoux) [inline src/nvim/ex_cmds_defs.h:198]: Here bools too.

File: src/nvim/search.h [hunk 6]
```diff
@@ -88,7 +88,7 @@ typedef struct searchstat
 {
     int     cur;      // current position of found words
     int     cnt;      // total count of found words
-    int     exact_match;    // TRUE if matched exactly on specified position
+    int     exact_match;    // true if matched exactly on specified position
```
Comment (vigoux) [inline src/nvim/search.h:91]: Should be a bool

**Review by vigoux — CHANGES_REQUESTED** [review]

File: src/nvim/buffer.c [hunk 7]
```diff
@@ -542,7 +542,7 @@ bool close_buffer(win_T *win, buf_T *buf, int action, bool abort_if_last)
 
   /*
    * Free all things allocated for this buffer.
-   * Also calls the "BufDelete" autocommands when del_buf is TRUE.
+   * Also calls the "BufDelete" autocommands when del_buf is true.
    */
```
Comment (vigoux) [inline src/nvim/buffer.c:546]: I think this comment needs to be changed to a `//`.

Comment (dundargoc) [reply src/nvim/buffer.c:546]: That's what I thought too. clint didn't give a warning so I just skipped it. Fixing it now.

File: src/nvim/buffer.c [hunk 8]
```diff
@@ -540,12 +540,10 @@ bool close_buffer(win_T *win, buf_T *buf, int action, bool abort_if_last)
...
-  /*
-   * Free all things allocated for this buffer.
-   * Also calls the "BufDelete" autocommands when del_buf is TRUE.
-   */
-  /* Remember if we are closing the current buffer.  Restore the number of
-   * windows, so that autocommands in buf_freeall() don't get confused. */
+  // Free all things allocated for this buffer.
+  // Also calls the "BufDelete" autocommands when del_buf is true.
+  // Remember if we are closing the current buffer.  Restore the number of
+  // windows, so that autocommands in buf_freeall() don't get confused. */
```
Comment (vigoux) [inline src/nvim/buffer.c:546]:
```suggestion
  // windows, so that autocommands in buf_freeall() don't get confused.
```

Comment (dundargoc) [reply src/nvim/buffer.c:546]: Shit, how did I miss that? Thank you.

File: src/nvim/mbyte.c [hunk 9]
```diff
@@ -2354,10 +2354,8 @@ int convert_setup(vimconv_T *vcp, char_u *from, char_u *to)
   return convert_setup_ext(vcp, from, true, to, true);
 }
 
-/*
- * As convert_setup(), but only when from_unicode_is_utf8 is TRUE will all
- * "from" unicode charsets be considered utf-8.  Same for "to".
- */
+// As convert_setup(), but only when from_unicode_is_utf8 is true will all
```
Comment (bfredl) [inline src/nvim/mbyte.c:2357]: three  `///` for function docstring

File: src/nvim/mouse.c [hunk 10]
```diff
@@ -571,7 +571,7 @@ static linenr_T find_longest_lnum(void)
 }
 
 ///
-/// Do a horizontal scroll.  Return TRUE if the cursor moved, FALSE otherwise.
+/// Do a horizontal scroll.  Return true if the cursor moved, false otherwise.
```
Comment (bfredl) [inline src/nvim/mouse.c:574]: `@Return` (and delete two empty comment lines)

File: src/nvim/mouse.c [hunk 11]
```diff
@@ -570,9 +570,7 @@ static linenr_T find_longest_lnum(void)
   return ret;
 }
 
-///
-/// Do a horizontal scroll.  Return TRUE if the cursor moved, FALSE otherwise.
-///
+/// Do a horizontal scroll.  @return true if the cursor moved, false otherwise.
```
Comment (vigoux) [inline src/nvim/mouse.c:573]: The `@return` should be on it's own line I think.

File: src/nvim/normal.c [hunk 12]
```diff
@@ -8146,9 +8146,7 @@ static void nv_event(cmdarg_T *cap)
   }
 }
 
-/*
- * Return TRUE when 'mousemodel' is set to "popup" or "popup_setpos".
- */
+/// @return true when 'mousemodel' is set to "popup" or "popup_setpos".
```
Comment (vigoux) [inline src/nvim/normal.c:8149]: Then the return type should be `bool` IMO.

File: src/nvim/search.h [hunk 13]
```diff
@@ -88,7 +88,7 @@ typedef struct searchstat
 {
     int     cur;      // current position of found words
     int     cnt;      // total count of found words
-    int     exact_match;    // TRUE if matched exactly on specified position
+    bool     exact_match;    // true if matched exactly on specified position
```
Comment (bfredl) [inline src/nvim/search.h:91]: alignment

**Comment by dundargoc** [comment]
I split up the work into multiple commits, I hope that's OK? If you want me to squash them later on then I'll do it, but I thought for reviewing purposes that it might be easier for me to split up the work into more logical pieces.

**Comment by bfredl** [comment]
I don't think we want a commit per file in this case :)

**Comment by dundargoc** [comment]
The changes where the scope is a function name is the case when e.g.

```
void dullify(int isPrime)
```

often span multiple files. If `isPrime` is changed to `bool` then all function calls to `dullify` should have and argument that is also `bool`.

**Comment by bfredl** [comment]
Sure, but it is _size_ of the commits that are relevant here :]

File: src/nvim/ex_getln.c [hunk 14]
```diff
@@ -166,7 +166,7 @@ typedef struct command_line_state {
   int firstc;
   long count;
   int indent;
-  int c;
+  bool c;
```
Comment (vigoux) [inline src/nvim/ex_getln.c:169]: This should remain an int, from what I saw in `command_line_execute` for example.

File: src/nvim/normal.c [hunk 15]
```diff
@@ -79,7 +79,7 @@ typedef struct normal_state {
   int mapped_len;
   int old_mapped_len;
   int idx;
-  int c;
+  bool c;
```
Comment (vigoux) [inline src/nvim/normal.c:82]: Here to, looking at the usage of this field afterwards, it should be kept as an int.

File: src/nvim/normal.c [hunk 16]
```diff
@@ -1340,7 +1340,7 @@ static int normal_check(VimState *state)
   quit_more = false;
 
   // If skip redraw is set (for ":" in wait_return()), don't redraw now.
-  // If there is nothing in the stuff_buffer or do_redraw is TRUE,
+  // If there is nothing in the stuff_buffer or do_redraw is true,
   // update cursor and redraw.
   if (skip_redraw || exmode_active) {
```
Comment (vigoux) [inline src/nvim/normal.c:1345]: @bfredl, this one is interesting, `exmode_active` is actually an int, that could be refactor as an enum.

**Review by vigoux — CHANGES_REQUESTED** [review]
There was some issues with ints turned to bools, while they should not.

File: src/nvim/ex_getln.c (hunk 14 above)
Comment (dundargoc) [reply src/nvim/ex_getln.c:169]: You are completely right, I had a brain fart.

File: src/nvim/normal.c (hunk 15 above)
Comment (dundargoc) [reply src/nvim/normal.c:82]: Brain fart #2

**Review by vigoux — APPROVED** [review]

**Pull request merged**